x-pack/solutions/observability/plugins/apm_data_access/server/utils/access_known_fields.ts

```diff
@@ -80,30 +99,43 @@ type ApmDocument<
  *
  * console.log(unflattened.service.name); // => outputs "node-svc" like above
  */
-export function accessKnownApmEventFields<
-  T extends Partial<FlattenedApmEvent>,
-  R extends keyof FlattenedApmEvent = never
->(fields: T, required?: R[]): ApmDocument<T, R>;
+export function accessKnownApmEventFields<T extends Partial<FlattenedApmEvent>>(
+  fields: T
+): ApmDocument<T, never>;
 
-export function accessKnownApmEventFields(fields: Record<string, any>, required?: string[]) {
-  if (required) {
-    ensureRequiredApmFields(fields, required);
-  }
+export function accessKnownApmEventFields(fields: Record<string, any>) {
```

Comment:
The API is changed here to not require validating immediately, preferring to defer validations with `requireFields` as part of a builder pattern approach.